Certain component, helper, and modifier imports fail to resolve with v2.0.0

Building with `staticHelpers`/`staticModifiers`/`staticComponents` set to `false` results in `@embroider/compat` failing to resolve `*.hbs.js` imports for certain template-only components, throwing a build error:
```
Build Error (PackageRunner) in components/maybe-in-element.js

Module not found: Error: Can't resolve '../node_modules/ember-maybe-in-element/components/maybe-in-element' in '$TMPDIR/embroider/40cd52/components/maybe-in-element.js'
```

Building with the same options set to `true` succeeds, but results in certain package components, helpers, and modifiers failing to resolve at runtime instead:
```
Uncaught Error: Attempted to resolve `ensure-safe-component`, which was expected to be a helper, but nothing was found.
    at resolveHelper (opcode-compiler.js:276:1)
    at encodeOp (opcode-compiler.js:2607:1)
    at pushOp (opcode-compiler.js:2521:1)
    at opcode-compiler.js:545:1
    at Compilers.compile (opcode-compiler.js:520:1)
    at expr (opcode-compiler.js:727:1)
    at Curry (opcode-compiler.js:972:1)
    at opcode-compiler.js:559:1
    at Compilers.compile (opcode-compiler.js:520:1)
    at expr (opcode-compiler.js:727:1)
```

I've had some success changing [compat/src/synthesize-template-only-components.ts#L10](https://github.com/embroider-build/embroider/blob/main/packages/compat/src/synthesize-template-only-components.ts#L10) to `['.hbs', '.hbs.js']` and looping over that in `crawl()`, although this only seems to fix the first issue

--- a/.gitignore
+++ b/.gitignore
@@ -65,3 +65,7 @@ tmp/
 
 # scenario tester debugging output
 tests/scenarios/output/
+
+# Sys files
+.DS_Store
+*.swp

--- a/packages/compat/src/synthesize-template-only-components.ts
+++ b/packages/compat/src/synthesize-template-only-components.ts
@@ -7,7 +7,7 @@ import { remove, outputFileSync, pathExistsSync } from 'fs-extra';
 const source = `import templateOnlyComponent from '@ember/component/template-only';
 export default templateOnlyComponent();`;
 
-const templateExtension = '.hbs';
+const templateExtensions = ['.hbs', '.hbs.js'];
 
 const jsExtensions = ['.js', '.ts', '.mjs', '.mts'];
 
@@ -60,12 +60,14 @@ function crawl(dir: string) {
   const seen = new Set<string>();
   if (pathExistsSync(dir)) {
     for (let file of walkSync(dir, { directories: false })) {
-      if (file.endsWith(templateExtension)) {
-        needed.add(file.slice(0, -1 * templateExtension.length));
-      } else {
-        const jsExtension = jsExtensions.find(ext => file.endsWith(ext));
-        if (jsExtension) {
-          seen.add(file.slice(0, -1 * jsExtension.length));
+      for (const templateExtension of templateExtensions) {
+        if (file.endsWith(templateExtension)) {
+          needed.add(file.slice(0, -1 * templateExtension.length));
+        } else {
+          const jsExtension = jsExtensions.find(ext => file.endsWith(ext));
+          if (jsExtension) {
+            seen.add(file.slice(0, -1 * jsExtension.length));
+          }
         }
       }
     }

